src/component/toolbox/feature/DataZoom.js

```diff
@@ -236,16 +240,30 @@ function updateZoomBtnStatus(featureModel, ecModel, view, payload, api) {
             zoomActive
             ? {
                 brushType: 'auto',
-                brushStyle: {
-                    // FIXME user customized?
-                    lineWidth: 0,
-                    fill: 'rgba(0,0,0,0.2)'
-                }
+                brushStyle: mapBrushStyle(featureModel.option.brushStyle)
             }
             : false
         );
 }
 
+function mapBrushStyle(brushStyle) {
+    var properties = [
+        ['fill', 'color'],
+        ['lineWidth', 'borderWidth'],
+        ['stroke', 'borderColor'],
+        ['opacity', 'opacity']
+    ];
+    var style = {};
+    for (var i = 0; i < properties.length; i++) {
+        var propName = properties[i][1];
+        var val = brushStyle[propName];
+        if (val != null) {
+            style[properties[i][0]] = val;
+        }
+    }
+    return style;
```

Comment:
firstly, you may consider es6 for..in syntax to simplify the for loop

secondly, you may consider es6 for..of syntax to shrink your for loop(reduce the number of variables and lines)

lastly, you can try out forEach to completely rewrite the function, i.e.:

```
[
     ['fill', 'color'],
     ['lineWidth', 'borderWidth'],
     ['stroke', 'borderColor'],
     ['opacity', 'opacity']
 ].forEach((item) =>{ console.log(item)})
```

Reply:
@chfw ECharts do not allow ES6 and ES5 syntax.

Reply:
Sorry, I am now aware of it.